src/oc/hocdec.h

```diff
@@ -140,17 +143,85 @@ struct Symbol { /* symbol table entry */
 
 using hoc_List = hoc_Item;
 
-union Datum { /* interpreter stack type */
-    double val;
-    Symbol* sym;
-    int i;
-    double* pval; /* first used with Eion in NEURON */
-    Object** pobj;
-    Object* obj; /* sections keep this to construct a name */
-    char** pstr;
-    hoc_Item* itm;
-    hoc_List* lst;
-    void* _pvoid; /* not used on stack, see nrnoc/point.cpp */
+namespace neuron::detail {
+struct _Datum { /* interpreter stack type */
+    union {
+        double val;
+        Symbol* sym;
+        int i;
+        double* pval; /* first used with Eion in NEURON */
+        Object** pobj;
+        Object* obj; /* sections keep this to construct a name */
+        char** pstr;
+        hoc_Item* itm;
+        hoc_List* lst;
+        void* _pvoid; /* not used on stack, see nrnoc/point.cpp */
+    };
+};
+}  // namespace neuron::detail
+
+struct Datum: private neuron::detail::_Datum {
+    template <typename T>
+    [[nodiscard]] T& literal_value() {
+        static_assert(std::is_standard_layout_v<T>);
```

Comment:
I think `std::is_trivially_copyable_v<T>` and `sizeof(T) <= sizeof(the union type)` is closer to the right constraint?

Reply:
Indeed, was thinking more along the C side.

Reply:
I think we should also add a trivially destructible requirement (as destructors are not called)